vm/rust/src/lib.rs

```diff
@@ -209,12 +209,45 @@ pub extern "C" fn cairoVMExecute(
                 JunoAppendGasConsumed(
                     reader_handle,
                     felt_to_byte_array(&t.actual_fee.0.into()).as_ptr(),
-                )
+                );
+
+                append_execution_info(
+                    reader_handle,
+                    t.into(),
+                );
             },
         }
     }
 }
 
+fn transaction_from_api(
+    tx: StarknetApiTransaction,
+    contract_class: Option<ContractClass>,
+) -> Result<Transaction, String> {
+    match tx {
+        StarknetApiTransaction::Deploy(deploy) => {
+            return Err(format!("Deploy transaction is not supported (transaction_hash={})", deploy.transaction_hash))
```

Comment:
This line is causing the following error even in cases where the transaction is not a deploy transaction. 
```
"error":{"code":40,"message":"Contract error","data":"Deploy transaction is not supported 
```

Looking at it, I realised that the error shows up in blocks that contain deploy transactions even though the transaction in question is not a Deploy transaction. 

Do we need to go through the whole block to get the transaction trace for a single transaction?

Reply:
> Do we need to go through the whole block to get the transaction trace for a single transaction?

Yes, because transactions get effected from state generated by the previous transactions in the same block. For trace to be accurate we need to execute the entire block

Reply:
Okay